GVFS/GVFS.Platform.Windows/WindowsPlatform.cs

```diff
@@ -266,6 +268,29 @@ public override bool TryInstallGitCommandHooks(GVFSContext context, string execu
             return true;
         }
 
+        public override bool TryVerifyAuthenticodeSignature(PhysicalFileSystem fileSystem, string path, out string subject, out string error)
+        {
+            using (Stream stream = fileSystem.OpenFileStream(path, FileMode.Open, FileAccess.Read, FileShare.Read, false))
+            using (PowerShell powershell = PowerShell.Create())
+            {
+                powershell.AddScript($"Get-AuthenticodeSignature -FilePath {path}");

```

Comment:
Is powershell the only way to accomplish this?  Did you try using X509Certificate?  Is there anyway to use a self-signed certificate with a subject that is expected in this code to get around these checks and pass the validation?

Reply:
I'd imagine if you used a self-signed certificate to sign a package that had the correct subject name, and had that or one of it's parent certificates in the trust chain installed in the computer's trusted store, then you could fool it. That would be entirely on the user/computer admin however.

You could also get a real certificate, whose certification path included one in the trusted store, that had the subject name "Microsoft Corporation" that wasn't actually "Microsoft Corporation" if there are rogue signing authorities who are shirking their responsibility to verify the details of the thing they're signing (those who end up in the untrusted store).

Reply:
@kewillford I did try `X509Certificate` - it does help verify whether the certificate is valid, but does not check whether the exe has been tampered with. Other options considered were pinvoke-ing into WinTrustAPI and Powershell command. Powershell was chosen because it is easier to use.

Tests that I've done - 
1. verify unsigned installers are not launched.
2. verify signed installers that have been tampered with are not launched.

>>I'd imagine if you used a self-signed certificate to sign a package that had the correct subject name, and had that or one of it's parent certificates in the trust chain installed in the computer's trusted store, then you could fool it. That would be entirely on the user/computer admin however.

Agree. Wondering if it would be an improvement to add a check for the issuer in addition to the signer. This would help avoid installing self-signed exes.